### ISSUE TITLE
no-invalid-block-param-definition false positives with {{#each}}

### ISSUE THREAD

**Issue by EvgenyOrekhov** [ember-template-lint/ember-template-lint#1193]
Code:

```hbs
<div>
  <MyComponent>
    {{#each this.foo as |bar|}}
      {{bar}}
    {{/each}}
  </MyComponent>
</div>
```

Expected result: no warnings.

Actual result:

```text
  1:0  error  Unexpected block usage in "<div>".   no-invalid-block-param-definition
  2:2  error  Unexpected block usage in "<MyComponent>". Missing " as " keyword before block symbol "|".  no-invalid-block-param-definition
```

ember-template-lint version: 2.2.0

**Comment by rwjblue** [comment]
This is very bizarre, can you share your configuration also?

**Comment by EvgenyOrekhov** [comment]
@rwjblue 

```js
"use strict";

module.exports = {
  extends: ["octane", "stylistic"],

  rules: {
    "inline-link-to": true,
    "deprecated-each-syntax": true,
    "deprecated-inline-view-helper": true,
    "no-whitespace-within-word": true,
    "attribute-indentation": true,
    "no-bare-strings": true,
    "no-yield-only": true,
    "no-whitespace-for-layout": true,
    "modifier-name-case": true,
    "no-heading-inside-button": true,
    "require-input-label": true,
    "no-invalid-link-title": true,

    "eol-last": "never",
    "no-curly-component-invocation": {
      noImplicitThis: true,
      requireDash: false,
      allow: ["app-version", "date-format"]
    },
    "template-length": { max: 200, min: 2 },

    "no-action": false,
    "no-action-modifiers": false,
    "no-element-event-actions": false,
    "no-passed-in-event-handlers": false,
    "no-invalid-block-param-definition": false
  }
};
```
